src/display.rs

```diff
@@ -78,61 +77,67 @@ impl Display for InlineTable {
     }
 }
 
-struct TableFormatState<'t> {
-    path: RefCell<Vec<&'t str>>,
-    table: Cell<&'t Table>,
-}
-
-impl<'a> Display for TableFormatState<'a> {
-    fn fmt(&self, f: &mut Formatter) -> Result {
-        let table = self.table.get();
+fn visit_nested_tables<'t, F>(
```

Comment:
Alex suggests that I should make `visit_nested_tables` be a method on Table, and that `callback` should be an object implementing a visitor trait.